src/components/block/index.ts

```diff
@@ -205,13 +205,18 @@ export default class Block extends EventsDispatcher<BlockEvents> {
    */
   private didMutated = _.debounce((mutationsOrInputEvent: MutationRecord[] | InputEvent = []): void => {
     const shouldFireUpdate = mutationsOrInputEvent instanceof InputEvent ||
-      !mutationsOrInputEvent.some(({
-        addedNodes = [],
-        removedNodes,
-      }) => {
-        return [...Array.from(addedNodes), ...Array.from(removedNodes)]
-          .some(node => $.isElement(node) && (node as HTMLElement).dataset.mutationFree === 'true');
-      });
+    !mutationsOrInputEvent.some(({
+      addedNodes = [],
+      removedNodes,
+      target,
+    }) => {
+      if ((target as HTMLElement).dataset?.mutationFree === 'true') {
+        return true;
+      }
+
+      return [...Array.from(addedNodes), ...Array.from(removedNodes)]
+        .some(node => $.isElement(node) && (node as HTMLElement).dataset.mutationFree === 'true');
+    });
```

Comment:
This code should be separated into several parts with clear names. For now, it is hard to read and understand.

Also, we need to write a tests for the covered cases.